src/stirling/source_connectors/socket_tracer/mux_table.h

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2018- The Pixie Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <map>
+
+#include "src/stirling/core/types.h"
+#include "src/stirling/source_connectors/socket_tracer/canonical_types.h"
+#include "src/stirling/source_connectors/socket_tracer/protocols/mux/types.h"
+
+namespace px {
+namespace stirling {
+
+// clang-format off
+static constexpr DataElement kMuxElements[] = {
+        canonical_data_elements::kTime,
+        canonical_data_elements::kUPID,
+        canonical_data_elements::kRemoteAddr,
+        canonical_data_elements::kRemotePort,
+        canonical_data_elements::kTraceRole,
+        {"req_type", "Mux message request type",
+         types::DataType::INT64,
+         types::SemanticType::ST_NONE,
+         types::PatternType::GENERAL_ENUM},
+        {"resp_type", "Mux message response type",
+         types::DataType::INT64,
+         types::SemanticType::ST_NONE,
+         types::PatternType::GENERAL_ENUM},
+        {"tag", "Mux message request tag",
+         types::DataType::INT64,
+         types::SemanticType::ST_NONE,
+         types::PatternType::GENERAL_ENUM},
```

Comment:
Is this tag used to match requests and responses. If so, do we need in the output table?

Reply:
Since mux multiplexes connections, I could see this being useful for differentiating when a service is making concurrent requests to the same backend instance (tag values would be different). This might be a niche use case, but I think it's worth keeping.

Reply:
We typically try to hide details that are not at the protocol's logical level. So for consistency with other protocols, I'd like to leave it out for now.

Because of issues with creating legacy, it's also easier to add it in later versus removing it.

Reply:
Sounds good :+1:. This is now removed from the mux table.